File: src/数字操作/q258_各位相加/Solution.java
package 数字操作.q258_各位相加;

/**
 * 找规律 o(1) xyz=100*x+10*y+z=99*x+9*y+x+y+z
 */
public class Solution {

    public int addDigits(int num) {
        if (num % 9 == 0 && num != 0) {
            num = 9;
        } else {
            num %= 9;
        }
        return num;
    }
}

File: src/数字操作/q258_各位相加/Solution.py
class Solution:
    def addAll(self,num:int) ->int:
        res = 0
        while num>0:
            res+=num%10
            num = num//10
            print(num)
        print(res)
        return res
    def addDigits(self, num: int) -> int:
        if num//10==0:
            return num
        res = self.addAll(num)
        print(res)
        while res//10!=0:
            res = self.addAll(res)
        return res
